datamapper/changeset.py

```diff
@@ -198,6 +272,28 @@ def change(self, changes: dict) -> Changeset:
         self._forced_changes = dict_merge(self._forced_changes, changes)
         return self
 
+    def put_change(self, field: str, value: Any) -> Changeset:
+        """
+        Puts a change on the given `field` with `value`.
+
+        The value is stored without validation.
+        """
+        self._forced_changes[field] = value
+        return self
+
+    def on_changed(
+        self, field: str, f: Callable[[Changeset, Any], Changeset]
+    ) -> Changeset:
+        """
+        If `field` has been changed, call callback function `f` with the value,
+        returning a new Changeset.
+        """
+        changed_value = self.fetch_change(field)
+        if changed_value is not None:
```

Comment:
I think this wouldn't behave the way I'd expect it to in the case where you'd be changing the value to `None`. This could be solved by:

```python
if self.has_change(field):
  return f(self, self.fetch_change(field))
else:
  return self
```

Reply:
Good call. Just made that change and added a test for it. Another bonus: it showed me that marshmallow was failing validations for null values, even if they aren't marked required 👍